crates/noirc_evaluator/src/ssa/ir/instruction/call.rs

```diff
@@ -165,6 +165,15 @@ pub(super) fn simplify_call(
                 SimplifyResult::None
             }
         }
+        Intrinsic::StrAsBytes => {
+            let str = dfg.get_array_constant(arguments[0]);
+            if let Some((str, typ)) = str {
+                let result = dfg.make_array(str, typ);
+                SimplifyResult::SimplifiedTo(result)
+            } else {
+                SimplifyResult::None
+            }
```

Comment:
```suggestion
            // Strings are already represented as bytes internally
            SimplifyResult::SimplifiedTo(arguments[0])
```
This is a bit of a weird one, but we can take advantage of the fact that strings are already represented as `&[u8]` internally. Your code is doing this as well by making an array with the same elements and type as the original array, but there's no need to make a new array in this case.